pkg/instancetype/revision/store_test.go

```diff
@@ -0,0 +1,559 @@
+			Spec: virtv1.VirtualMachineInstanceSpec{
+				Domain: virtv1.DomainSpec{},
+			},
+		}
+		vm.Namespace = k8sv1.NamespaceDefault
+
+		_, err := virtClient.VirtualMachine(vm.Namespace).Create(context.Background(), vm, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("store instancetype", func() {
+		It("store returns error when instancetypeMatcher kind is invalid", func() {
+			vm.Spec.Instancetype = &virtv1.InstancetypeMatcher{
+				Kind: "foobar",
+			}
+			Expect(storeHandler.Store(vm)).To(MatchError(ContainSubstring("got unexpected kind in InstancetypeMatcher")))
+		})
+
+		It("store returns nil when no instancetypeMatcher is specified", func() {
+			vm.Spec.Instancetype = nil
+			Expect(storeHandler.Store(vm)).To(Succeed())
+		})
+
+		Context("using global ClusterInstancetype", func() {
+			var clusterInstancetype *instancetypev1beta1.VirtualMachineClusterInstancetype
+
+			BeforeEach(func() {
+				clusterInstancetype = &instancetypev1beta1.VirtualMachineClusterInstancetype{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "VirtualMachineClusterInstancetype",
+						APIVersion: instancetypev1beta1.SchemeGroupVersion.String(),
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       "test-cluster-instancetype",
+						UID:        resourceUID,
+						Generation: resourceGeneration,
+					},
+					Spec: instancetypev1beta1.VirtualMachineInstancetypeSpec{
+						CPU: instancetypev1beta1.CPUInstancetype{
+							Guest: uint32(2),
+						},
+						Memory: instancetypev1beta1.MemoryInstancetype{
+							Guest: resource.MustParse("128Mi"),
+						},
+					},
+				}
+
+				_, err := virtClient.VirtualMachineClusterInstancetype().Create(
+					context.Background(), clusterInstancetype, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				err = clusterInstancetypeInformerStore.Add(clusterInstancetype)
+				Expect(err).ToNot(HaveOccurred())
+
+				vm.Spec.Instancetype = &virtv1.InstancetypeMatcher{
+					Name: clusterInstancetype.Name,
+					Kind: apiinstancetype.ClusterSingularResourceName,
+				}
+
+				_, err = virtClient.VirtualMachine(vm.Namespace).Update(context.Background(), vm, metav1.UpdateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("store VirtualMachineClusterInstancetype ControllerRevision", func() {
+				clusterInstancetypeControllerRevision, err := revision.CreateControllerRevision(vm, clusterInstancetype)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(storeHandler.Store(vm)).To(Succeed())
+				Expect(vm.Spec.Instancetype.RevisionName).To(Equal(clusterInstancetypeControllerRevision.Name))
+
+				createdCR, err := virtClient.AppsV1().ControllerRevisions(vm.Namespace).Get(
+					context.Background(), vm.Spec.Instancetype.RevisionName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(createdCR).To(Equal(clusterInstancetypeControllerRevision))
+			})
+
+			It("store returns a nil revision when RevisionName already populated", func() {
```

Comment:
Would it make more sense to name this something like existing revision names are not replaced? You never check for a nil return.